lib/utils/config.ts

```diff
@@ -66,27 +67,30 @@ export function generateBaseConfig(
 
 export function generateApplicationConfig(
 	application: ResinSdk.Application,
-	options: {},
+	options: { version?: string },
 ) {
-	return generateBaseConfig(application, options).tap(config =>
-		addApplicationKey(config, application.id),
-	);
+	return generateBaseConfig(application, options).tap(config => {
+		if (semver.satisfies(options.version, '>=2.7.8')) {
+			return addProvisioningKey(config, application.id);
+		} else {
+			return addApplicationKey(config, application.id);
+		}
+	});
 }
 
 export function generateDeviceConfig(
-	device: ResinSdk.Device & { application_name: string },
+	device: ResinSdk.Device & { belongs_to__application: ResinSdk.PineDeferred },
```

Comment:
We might need to change this once its consumers get rewritten to TS.
It would be great to check both `__id` & `[0].id` like what `getId` does.

Reply:
Yeah, once we've got more TS in here and consistent usage of the same SDK Version everywhere, there's a lot of improvements like this (and your other comment) I'd like to do.

For now I think this is ok though - even if we need to change it later, it'll be something the type system finds and complains about for us, not a bug.